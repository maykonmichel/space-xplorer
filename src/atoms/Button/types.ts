import {ViewStyle} from 'react-native';

export type Color = 'secondary' | 'light';

export type Variant = 'outlined' | 'text';

export type Props = {
  accessibilityLabel?: string | undefined;
  color?: Color;
  onPress: () => void;
  style?: ViewStyle;
  title: string;
  variant?: Variant;
};
