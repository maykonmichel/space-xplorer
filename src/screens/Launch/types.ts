export type LaunchParams = {
  id: string;
};
