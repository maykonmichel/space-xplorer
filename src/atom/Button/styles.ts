import {StyleSheet} from 'react-native';

import colors from '~/theme/colors';

export default StyleSheet.create({
  container: {
    borderRadius: 4,
    borderWidth: 1,
    borderColor: colors.secondary.main,
    paddingHorizontal: 16,
    paddingVertical: 8,
  },
  title: {
    color: colors.secondary.main,
    textTransform: 'uppercase',
    fontWeight: 'bold',
    textAlign: 'center',
  },
});
