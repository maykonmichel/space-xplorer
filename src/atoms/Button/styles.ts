import {StyleSheet} from 'react-native';

import colors from '~/theme/colors';

import {Color, Props} from './types';

const getHex = (color?: Color) => {
  switch (color) {
    case 'light':
      return colors.neutral.light;
    case 'secondary':
      return colors.secondary.main;
    case 'primary':
    default:
      return colors.primary.main;
  }
};

export const useStyles = ({color, variant}: Props) => {
  const hex = getHex(color);

  return StyleSheet.create({
    container: {
      borderRadius: 4,
      borderWidth: variant === 'text' ? 0 : 1,
      borderColor: hex,
      paddingHorizontal: 16,
      paddingVertical: 8,
    },
    title: {
      color: hex,
      textTransform: 'uppercase',
      fontWeight: 'bold',
      textAlign: 'center',
    },
  });
};
