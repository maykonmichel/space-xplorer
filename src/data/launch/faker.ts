import faker from 'faker';
import {repeat} from 'ramda';

import {Launch} from '~/data/launch';

export const fakerFlickrImages = (): Launch['links']['flickr_images'] =>
  repeat(faker.image.image(), faker.datatype.number({min: 1, max: 5}));

export const fakerLaunch = (): Launch => ({
  id: faker.datatype.number().toString(),
  details: faker.lorem.paragraph(),
  mission_name: faker.lorem.words(),
  launch_date_formatted: faker.date.past().toLocaleDateString(),
  launch_date_local: faker.date.past().toLocaleDateString(),
  links: {
    article_link: faker.internet.url(),
    flickr_images: fakerFlickrImages(),
  },
  rocket: {
    rocket_name: faker.lorem.words(),
  },
});
