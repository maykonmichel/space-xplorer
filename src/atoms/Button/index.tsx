import React, {FC} from 'react';
import {ButtonProps, Text, TouchableOpacity, ViewStyle} from 'react-native';

import styles from './styles';

export type Props = ButtonProps & {
  style?: ViewStyle;
};

const Button: FC<Props> = ({title, style, ...props}) => {
  return (
    <TouchableOpacity
      accessibilityRole={'button'}
      style={[styles.container, style]}
      {...props}>
      <Text style={styles.title}>{title}</Text>
    </TouchableOpacity>
  );
};

export default Button;
