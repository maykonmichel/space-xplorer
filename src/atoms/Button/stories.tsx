import {action} from '@storybook/addon-actions';
import {select, text} from '@storybook/addon-knobs';
import {storiesOf} from '@storybook/react-native';
import React from 'react';

import Button from '.';

storiesOf('Button', module).add('sandbox', () => (
  <Button
    title={text('title', 'Title')}
    onPress={action('clicked-text')}
    color={select('color', ['primary', 'secondary', 'light'], 'primary')}
    variant={select('variant', ['outlined', 'text'], 'outlined')}
  />
));
