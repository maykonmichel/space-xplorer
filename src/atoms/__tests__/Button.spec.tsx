import {fireEvent, render} from '@testing-library/react-native';
import faker from 'faker';
import React from 'react';

import Button from '~/atoms/Button';
import {Props} from '~/atoms/Button/types';

const getProps = (): Props => ({
  title: faker.lorem.words(),
  onPress: jest.fn(),
});

describe('<Button />', () => {
  it('should render as expected', () => {
    const props = getProps();
    const {toJSON} = render(<Button {...props} title={'Test button'} />);

    expect(toJSON()).toMatchSnapshot();
  });

  it('should show title', () => {
    const props = getProps();
    const {queryByText} = render(<Button {...props} />);

    const {title} = props;

    expect(queryByText(title)).toBeTruthy();
  });

  it('should emit press event', () => {
    const props = getProps();
    const {getByRole} = render(<Button {...props} />);

    const button = getByRole('button');

    fireEvent.press(button);

    const {onPress} = props;

    expect(onPress).toBeCalled();
  });
});
