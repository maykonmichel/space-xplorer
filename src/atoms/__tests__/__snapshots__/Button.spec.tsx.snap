// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render as expected 1`] = `
<View
  accessibilityRole="button"
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderColor": "#EC407A",
      "borderRadius": 4,
      "borderWidth": 1,
      "opacity": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 8,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#EC407A",
        "fontWeight": "bold",
        "textAlign": "center",
        "textTransform": "uppercase",
      }
    }
  >
    Test button
  </Text>
</View>
`;
